### PR TITLE
fix(notifications): validate offset and limit query params to prevent NaN errors

### PR DESCRIPTION
**Pull request by suhr25**
### SUMMARY

This PR adds proper validation for `offset` and `limit` query params in the notifications API so invalid values don’t reach the database.
It updates `listNotificationsHandler()` in `plugins/notifications-backend/src/service/router.ts`.

---

### FIX

**Before:**

```ts
if (req.query.offset) {
  opts.offset = Number.parseInt(req.query.offset.toString(), 10);
}
if (req.query.limit) {
  opts.limit = Number.parseInt(req.query.limit.toString(), 10);
}
```

**After:**

```ts
if (req.query.offset) {
  const offset = Number.parseInt(req.query.offset.toString(), 10);
  if (isNaN(offset)) {
    throw new InputError('offset must be a valid integer');
  }
  opts.offset = offset;
}
if (req.query.limit) {
  const limit = Number.parseInt(req.query.limit.toString(), 10);
  if (isNaN(limit)) {
    throw new InputError('limit must be a valid integer');
  }
  opts.limit = limit;
}
```

---

### VERIFICATION

Requests like `/api/notifications?limit=foo` or `offset=bar` now return a clean `400` with a clear error instead of a DB failure.
Normal requests (`limit=10`, `offset=0`) behave exactly the same as before.

### CODE REVIEW

File: plugins/notifications-backend/src/service/router.ts [hunk 1]
```diff
@@ -497,10 +497,18 @@ export async function createRouter(
...
+        throw new InputError('offset must be a valid integer');
+      }
+      opts.offset = offset;
     }
     if (req.query.limit) {
-      opts.limit = Number.parseInt(req.query.limit.toString(), 10);
+      const limit = Number.parseInt(req.query.limit.toString(), 10);
+      if (isNaN(limit)) {
+        throw new InputError('limit must be a valid integer');
+      }
```
[the remark this answers is not in the document]
Comment (hudsonb) [reply plugins/notifications-backend/src/service/router.ts]: +1

File: plugins/notifications-backend/src/service/router.ts [hunk 2]
```diff
@@ -497,10 +497,18 @@ export async function createRouter(
...
+      opts.limit = limit;
     }
```
[the remark this answers is not in the document]
Comment (hudsonb) [reply plugins/notifications-backend/src/service/router.ts:512]: +1, suggest adding related tests

[the remark this answers is not in the document]
Comment (hudsonb) [reply plugins/notifications-backend/src/service/router.ts:512]: +1, seems worth capturing these additional edge cases while addressing this

**Review by hudsonb — APPROVED** [review]

**Comment by suhr25** [comment]
Hi @hudsonb,
Is it ready to merge??
Thanks for your time!

**Review by hudsonb — APPROVED** [review]

**Comment by suhr25** [comment]
Hi @hudsonb,
Thanks for approving the PR!
It looks like a code owner review is still pending. Could you please guide on the next steps or tag the appropriate reviewer?
Appreciate your help!
